pkg/pillar/cmd/volumemgr/handledownloader.go

```diff
@@ -25,17 +28,37 @@ func AddOrRefcountDownloaderConfig(ctx *volumemgrContext, status types.VolumeSta
 	} else {
 		log.Debugf("AddOrRefcountDownloaderConfig: add for %s\n",
 			status.BlobSha256)
+
+		name := status.DownloadOrigin.Name
+
+		// where should the final downloaded file be?
+		locFilename := path.Join(types.DownloadDirname, status.ObjType, "pending", status.VolumeID.String(), path.Base(name))
+		// try to reserve storage, must be released on error
+		size := status.DownloadOrigin.MaxSizeBytes // XXX should this be MaxSize
+		kb := types.RoundupToKB(size)
+		if ret, errStr := tryReserveSpace(ctx, &status, kb, name); !ret {
+			// *** THIS NEEDS TO BE FIXED
+			// it is still the old one from downloader
+			// instead, we need our own retry loop here on space
+			status.RetryCount++
+			publishVolumeStatus(ctx, &status)
+			// *** END NEEDS TO BE FIXED
+			log.Errorf("AddOrRefcountDownloaderConfig(%s): deferred with %s", name, errStr)
+			return
+		}
+
```

Comment:
This feels a bit misplaced in this function which is instructing the downloader to download.
Shouldn't the size enforcement be on the volumes i.e., a check in handleCreate in volumemgr to determine whether there is space for the volume?

Reply:
Honestly? I am not sure. This was originally done in the downloader as soon as it received the config, so I moved it up into volumemgr. 

If it is supposed to be on the volumes, why was it in downloader before? And doesn't it need to reserve the space just to ensure it has enough space to download?

Reply:
It was in downloader and then partly in verifier plus some checks in zedmanager; system has evolved and this check is lagging.

But need to check for a volume even if no download is needed e.g., create 3G empty volume. Hence handleCreate makes the most sense.

Reply:
This whole piece, though, was just to get `tryReserveSpace()`. As that is now (potentially) removed, it might not matter. I will await your feedback on that commit.